base/windows/battle/BattleStatusWindow.ts

```diff
@@ -0,0 +1,511 @@
+        let effects:{stat: string, values: number[]}[] = [];
+
+        let base_stats = [main_stats.ATTACK, main_stats.DEFENSE, main_stats.AGILITY];
+
+        for(let index in base_stats){
+            let effect = {stat: base_stats[index], values: [this.selected_char[base_stats[index]] - 
+                this.selected_char.preview_stat_without_abilities_effect(base_stats[index])]};
+            effects.push(effect);
+        }
+
+        let elemental_base = this.selected_char.preview_elemental_stats_without_abilities_effect();
+
+        let resist_changes = [];
+        resist_changes.push(this.selected_char.venus_resist_current - elemental_base[elements.VENUS].resist);
+        resist_changes.push(this.selected_char.mercury_resist_current - elemental_base[elements.MERCURY].resist);
+        resist_changes.push(this.selected_char.mars_resist_current - elemental_base[elements.MARS].resist);
+        resist_changes.push(this.selected_char.jupiter_resist_current - elemental_base[elements.JUPITER].resist);
+
+        let power_changes = [];
+        power_changes.push(this.selected_char.venus_power_current - elemental_base[elements.VENUS].power);
+        power_changes.push(this.selected_char.mercury_power_current - elemental_base[elements.MERCURY].power);
+        power_changes.push(this.selected_char.mars_power_current - elemental_base[elements.MARS].power);
+        power_changes.push(this.selected_char.jupiter_power_current - elemental_base[elements.JUPITER].power);
+
+        if(resist_changes.reduce((a, b) => a + b, 0) != 0)
+            effects.push({stat: "res", values: resist_changes});
+        if(power_changes.reduce((a, b) => a + b, 0) != 0)
+            effects.push({stat: "pow", values: power_changes});
+
+        return effects;
+    }
+
+    private get_status_effects(menu?:boolean){
+        if(menu){
+            return _.sortBy([...this.data.info.main_char_list[this.selected_char.key_name].permanent_status],
+                s => ordered_status_menu.indexOf(s));
+        }
+        else{
+            return _.sortBy([...this.data.info.main_char_list[this.selected_char.key_name].temporary_status].
+                concat([...this.data.info.main_char_list[this.selected_char.key_name].permanent_status]),
+                s => ordered_status_battle.indexOf(s));
+        }
+    }
+
+    private set_sprites(){
+        if(this.battle_sprite) this.battle_sprite.destroy();
+        if(this.avatar) this.avatar.destroy();
+
+        if(this.effect_sprites.length > 0){
+            for(let index in this.effect_sprites){
+                this.effect_sprites[index].destroy();
+            }
+            this.effect_sprites = [];
+        }
+
+        this.avatar = this.window.create_at_group(BattleStatusWindow.AVATAR.X, BattleStatusWindow.AVATAR.Y,
+            "avatars", undefined, this.selected_char.key_name, BattleStatusWindow.GROUP_KEY);
+        
+        let sprite_key = this.selected_char.key_name + "_battle";
```

Comment:
Please change this line to `sprite_base.getActionKey(base_actions.BATTLE)`. (there's no `BATTLE` in base_actions, please add).